base/linalg/symmetric.jl

```diff
@@ -41,6 +41,15 @@ julia> Slower = Symmetric(A, :L)
 Note that `Supper` will not be equal to `Slower` unless `A` is itself symmetric (e.g. if `A == A.'`).
 """
 Symmetric(A::AbstractMatrix, uplo::Symbol=:U) = (checksquare(A); Symmetric{eltype(A),typeof(A)}(A, char_uplo(uplo)))
+Symmetric(A::Symmetric) = A
+function Symmetric(A::Symmetric, uplo::Symbol)
+    if A.uplo == char_uplo(uplo)
+        return A
+    else
+        throw(ArgumentError("Cannot construct Symmetric; uplo doesn't match"))
```

Comment:
I think it is unlikely that people will hit this exception on purpose, i.e. where they want to flip the storage. We can revisit if it is really an issue.